manageiq-operator/pkg/helpers/miq-components/util.go

```diff
@@ -52,3 +56,96 @@ func addResourceReqs(memLimit, memReq, cpuLimit, cpuReq string, c *corev1.Contai
 
 	return nil
 }
+
+func serviceMutateFn(service *corev1.Service, labels map[string]string, ports []corev1.ServicePort, selector map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		service.ObjectMeta.Labels = labels
+		service.Spec.Ports = ports
+		service.Spec.Selector = selector
+		return nil
+	}
+
+	return mutateFn
+}
+
+func secretMutateFn(secret *corev1.Secret, labels map[string]string, data map[string][]byte, stringData map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		secret.ObjectMeta.Labels = labels
+		if len(data) > 0 {
+			secret.Data = data
+		}
+		if len(stringData) > 0 {
+			secret.StringData = stringData
+		}
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func configMapMutateFn(configMap *corev1.ConfigMap, labels map[string]string, data map[string]string) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		configMap.ObjectMeta.Labels = labels
+		if len(data) > 0 {
+			configMap.Data = data
+		}
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func pvcMutateFn(pvc *corev1.PersistentVolumeClaim, labels map[string]string, accessModes []corev1.PersistentVolumeAccessMode, resources corev1.ResourceRequirements) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		pvc.ObjectMeta.Labels = labels
+		pvc.Spec.AccessModes = accessModes
+		pvc.Spec.Resources = resources
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func roleMutateFn(role *rbacv1.Role, rules []rbacv1.PolicyRule) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		role.Rules = rules
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func roleBindingMutateFn(roleBinding *rbacv1.RoleBinding, roleRef rbacv1.RoleRef, subjects []rbacv1.Subject) controllerutil.MutateFn {
+
+	mutateFn := func() error {
+		roleBinding.RoleRef = roleRef
+		roleBinding.Subjects = subjects
+		return nil
+	}
+
+	return mutateFn
+
+}
+
+func deploymentMutateFn(deployment *appsv1.Deployment, repNum *int32, container corev1.Container, podSpec corev1.PodSpec, strategy appsv1.DeploymentStrategyType) controllerutil.MutateFn {
```

Comment:
Generally I feel like this isn't worth splitting out into a separate function.  Passing 5 parameters to a function that only has 8 real lines doesn't feel like it's making anything better. I would rather see the mutate function defined for each deployment in their own methods.